src/strands/tools/mcp/mcp_client.py

```diff
@@ -63,17 +63,23 @@ class MCPClient:
     from MCP tools, it will be returned as the last item in the content array of the ToolResult.
     """
 
-    def __init__(self, transport_callable: Callable[[], MCPTransport]):
+    def __init__(self, transport_callable: Callable[[], MCPTransport], server_startup_timeout: int = 30):
         """Initialize a new MCP Server connection.
 
         Args:
             transport_callable: A callable that returns an MCPTransport (read_stream, write_stream) tuple
+            server_startup_timeout: Timeout after which MCP server initialization should be cancelled
+                Defaults to 30.
         """
+        self.server_startup_timeout = server_startup_timeout
```

Comment:
Let's make this private until we determine that it needs to be public

```suggestion
        self._server_startup_timeout = server_startup_timeout
```